client/src/components/messenger/Conversations.js

```diff
@@ -1,53 +1,112 @@
-import React from "react";
+import {useEffect, useState, React} from "react";
+import { useSelector } from "react-redux";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import Dialog from "@material-ui/core/Dialog";
 
+import plus from "../../assets/plus.svg";
 import ConversationItem from "./ConversationItem";
+import { ConversationsController } from "./controllers/Conversations";
+import { ContactsController } from "./controllers/Contacts";
+
+const AddConversationDialog = ({open, onClose, token}) => {
+    const [list, setList] = useState([]);
+    const [contacts, setContacts] = useState([]);
+    const [init, setInit] = useState(false);
+
+    const handleClose = () => onClose(list);
+    const handleChange = (e) => {
+        // if checkbox is checked then add it to the list else remove it from the list
+        setList(e.target.checked ? [...list, e.target.value] : list.filter(item => item !== e.target.value));
+    };
+
+    useEffect(() => {
+        if (!init) { // if this is the first time rendering, get user contacts from server
+            (async () => setContacts(await ContactsController.getContacts( token )))();
+            setInit(true);
+        }
+    }, [init]);
+
+    return (
+        <Dialog onClose={handleClose} aria-labelledby="simple-dialog-title" open={open}>
+            <DialogTitle id="simple-dialog-title">Create a conversation</DialogTitle>
+            <ul className="add_participants">
+                {contacts.map(contact =>
+                    <li key={contact.contactId}>
+                        <img src={contact.contactInfo.imageUrl} alt="contact_img"/>
+                        <span>{contact.contactInfo.displayName}</span>
+                        <input
+                            type="checkbox"
+                            name="participant"
+                            value={contact.contactId}
+                            onChange={handleChange}
+                        />
+                    </li>
+                )}
+            </ul>
+            <button onClick={() => handleClose()} name="create" className="addContact">Create</button>
+        </Dialog>
+    );
+};
+
+
+const Conversations = ({selectConversation}) => {
+    const [dialogOpen, setDialogOpen] = useState(false);
+    const [conversations, setConversations] = useState([]);
+    const [init, setInit] = useState(false);
+
+    // Get the token from the users global state.
+    const token = useSelector((state) => state.user.token);
+
+    useEffect(() => {
+        // if this is the first time rendering, get user conversations from server
+        if (!init) {
+            (async () => setConversations(await ConversationsController.getConversations(token)))();
```

Comment:
I know why you're doing this immediately executing async lambda, but I don't love it 😄